launcher/src/main/java/com/airbnb/mvrx/launcher/MvRxLauncherMockActivity.kt

```diff
@@ -0,0 +1,197 @@
+         * It is an error to call this if a mock is not set.
+         *
+         * @param showView This callback must present the given view (eg if it's a Fragment, add it to the activity).
+         * The implementation will depend on the type of [MvRxView].
+         * It is recommended that the view be added synchronously so that this can catch and handle
+         * errors gracefully if there is an exception thrown when it is shown. Otherwise the launcher
+         * may get into an infinite loop as it restarts and tries to show the same mock again automatically.
+         *
+         * @param onFailure Called if an error is caught while the view is being shown. Use this to
+         * gracefully recover and exit from where you are showing the mock.
+         *
+         * @see showNextMockFromActivity
+         */
+        fun showNextMock(
+            showView: (MvRxView) -> Unit,
+            onFailure: (Throwable) -> Unit = { throw it }
+        ) {
+            val mock = nextMockToShow ?: error("Mock was not set")
+
+            val mockedView: MockedView<*> = mock.createView(mockBehavior(mock.mock))
+
+            val view = mockedView.viewInstance
+            view.lifecycle.addObserver(ViewModelEnabler(mockedView, mock))
+
+            @Suppress("Detekt.TooGenericExceptionCaught")
+            try {
+                showView(view)
+            } catch (e: Throwable) {
+                Log.e(
+                    "MvRx Launcher",
+                    "${view.javaClass.simpleName} crashed while opening.",
+                    e
+                )
+                onFailure(e)
+            }
+        }
+
+        /**
+         * Similar to [showNextMock], but with a few common defaults added for the case
+         * where a view is shown from an Activity.
+         */
+        fun showNextMockFromActivity(
+            activity: Activity,
+            showView: Activity.(MvRxView) -> Unit
+        ) {
+            val mock = nextMockToShow ?: error("Mock was not set")
+
+            val mockedView: MockedView<*> = mock.createView(mockBehavior(mock.mock))
+
+            val view = mockedView.viewInstance
+            view.lifecycle.addObserver(ViewModelEnabler(mockedView, mock))
+
+            @Suppress("Detekt.TooGenericExceptionCaught")
+            try {
+                activity.showView(view)
+
+                if (activity.intent.getBooleanExtra(FINISH_AFTER_LAUNCH, false)) {
+                    // If we are initializing from arguments then network requests will be
+                    // made, and we want to give enough time for them to complete to make
+                    // sure the parsing and display it tested as well.
+                    val isInitializing =
+                        mock.mock.isDefaultInitialization || mock.mock.isForProcessRecreation
+                    val finishAfterMs: Long = if (isInitializing) 3000 else 500
```

Comment:
I also wonder if this can leverage some of the TTI concepts we use -- i.e. waiting until async props marked as necessary for load go to a terminal state.

Reply:
those are nice  ideas for follow up work  :+1: